impl/src/main/java/uk/ac/manchester/cs/owl/owlapi/Internals.java

```diff
@@ -877,8 +878,18 @@ public boolean isEmpty() {
      * @return set of values
      */
     @Nonnull
-    public <K> Collection<OWLAxiom> filterAxioms(
+    public <K> Collection<? extends OWLAxiom> filterAxioms(
             @Nonnull OWLAxiomSearchFilter filter, @Nonnull K key) {
+        if(filter == Filters.annotations) {
```

Comment:
This would be a little easier to parse if the constants in Filters were in upper-case, ie, Filters.ANNOTATIONS, but I guess changing that in version 4 is probably out of the question now.